### PR TITLE
[1.2.0] Implement ExecuteOnKey map method

### LINKED ISSUE #624: ExecuteOnKey function is missing in the new API
Go version: 1.17
Hazelcast Go Client version: 1.0.0
Hazelcast server version: 4.x
Number of the clients: - 
Cluster size, i.e. the number of Hazelcast cluster members: - 
OS version : Linux

#### Expected behaviour
hazelcast.Map has ExecuteOnKey function to run entry processor on the specific key in the map

#### Actual behaviour
The method is missing, only ExecuteOnEntries method is available

#### Steps to reproduce the behaviour

ExecuteOnKey method is used in several places in my code and it worked fine on the previous client version.

### PR DESCRIPTION
**Pull request by JorgenPo**
Resolves #624

### CODE REVIEW

**Comment by yuce** [comment]
Hi @JorgenPo Thanks for your PR! Could you also add a test to the PR?

File: proxy_map.go [hunk 1]
```diff
@@ -272,6 +272,28 @@ func (m *Map) ExecuteOnEntries(ctx context.Context, entryProcessor interface{})
...
+func (m *Map) ExecuteOnKey(ctx context.Context, entryProcessor interface{}, key interface{}) (interface{}, error) {
+	processorData, err := m.validateAndSerialize(entryProcessor)
+	if err != nil {
+		return nil, err
+	}
+	keyData, err := m.validateAndSerialize(key)
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (yuce) [inline proxy_map.go:285]: This blank line is not necessary.

File: proxy_map.go [hunk 2]
```diff
@@ -272,6 +272,28 @@ func (m *Map) ExecuteOnEntries(ctx context.Context, entryProcessor interface{})
...
+	lid := extractLockID(ctx)
+
```
Comment (yuce) [inline proxy_map.go:287]: This blank line is not necessary.

**Review by yuce — CHANGES_REQUESTED** [review]

**Comment by JorgenPo** [comment]
> Hi @JorgenPo Thanks for your PR! Could you also add a test?

Sure, I'll try to add a test for the function

**Comment by yuce** [comment]
Thanks! I've added a test for `ExecuteOnEntries`. The same setup can also be used for `ExecuteOnKey`.

**Comment by yuce** [comment]
Would it be possible for you to merge https://github.com/hazelcast/hazelcast-go-client/pull/630/files into your PR, so you can immediately access that setup?

**Comment by JorgenPo** [comment]
> Would it be possible for you to merge https://github.com/hazelcast/hazelcast-go-client/pull/630/files into your PR, so you can immediately access that setup?

I merged your commit and added a test for ExecuteOnKey

File: proxy_map.go (hunk 2 above)
Comment (JorgenPo) [reply proxy_map.go]: Fixed this

**Comment by yuce** [comment]
Thanks @JorgenPo, we'll merge this after releasing v1.1.1

**Comment by yuce** [comment]
Hi @JorgenPo I'm about to merge this PR, but could you sign the CLA and resolve the conflict in `map_it_test.go`?

**Comment by utku-caglayan** [comment]
@JorgenPo sadly we need to continue this wok on another PR since we need to provide it on v1.2.0. Sorry for the inconvenience. Your contributions are always welcome!

**Pull request closed without merging**